cypress/e2e/requests.cy.js

```diff
@@ -2,50 +2,102 @@ describe('Viewing all requests', () => {
       // Visit a protected route in order to allow cypress to set the cookie and mock the login
-      cy.visit("/requests")
+      cy.visit('/requests')
       cy.get('div.alert-heading').contains('Unauthorized').then(() => {
-        cy.log("A logged out user is not able to view requests.")
+        cy.log('A logged out user is not able to view requests.')
       })
     })
   })
   
   describe('as a logged in user', () => {
     let scientistApiBaseURL = `https://${Cypress.env('NEXT_PUBLIC_PROVIDER_NAME')}.scientist.com/api/v2`
+    // declare variables that can be used to change how the response is intercepted.
     let requestList
+    let loading
+    let error
 
     beforeEach(() => {
       // Call the custom cypress command to log in
       cy.login(Cypress.env('TEST_SCIENTIST_USER'), Cypress.env('TEST_SCIENTIST_PW'))
-
       // Intercept the response from the endpoint to view all requests
+      // TODO(summer-cook): extract out this base url into the config to use as an environment variable. it was not cooperating before
       cy.intercept('GET', `${scientistApiBaseURL}/quote_groups/mine.json`, (req) => {
-        if (requestList === true) {
-          req.reply({fixture: 'all-requests/requests.json'})
-        } else {
-          req.reply({fixture: 'all-requests/no-requests.json'})
+        if ((requestList === undefined) && (loading === true)) {
+          // reply with an empty response: both data and error will be undefined.
+          req.reply()
+        } else if ((requestList === undefined) && (loading === undefined) && (error === true)) {
+          // error will be defined
+          req.reply({ statusCode: 404 })
+        } else if (requestList === true) {
+          // reply with a request body- default status code is 200
+          req.reply({ fixture: 'all-requests/requests.json' })
+        } else if (requestList === false) {
+          req.reply({ fixture: 'all-requests/no-requests.json' })
         }
-      }).as('useAllRequests')
-      cy.visit("/requests")
+      })
+      // Intercept the response from the endpoint that gets the default ware ID
+      cy.intercept('GET', `${scientistApiBaseURL}/wares.json?q=make-a-request`, { fixture: 'all-requests/make-a-request.json' })
+      cy.visit('/requests')
     })
 
-    context('has requests', () => {
+
+    context('request list is loading', () => {
       before(() => {
-        requestList = true
+        loading = true
       })
-      it("shows the user's request list.", () => {
-        cy.get('article.request-item').should('exist').then(() => {
-          cy.log('Successfully viewing request list.')
+      it('should show a loading spinner.', () => {
+        cy.get("[aria-label='tail-spin-loading']").should('be.visible').then(() => {
+          cy.log('Loading spinner displays correctly.')
         })
       })
     })
 
-    context('has 0 requests', () => {
+    context('error while making a request to the api', () => {
```

Comment:
why is the api throwing an error for a logged in user?

Reply:
the actual api is not throwing an error, we are just mocking the case where the api throws a 404. this is testing the case where the error component renders for [this ticket ](https://github.com/orgs/scientist-softserv/projects/27/views/1?pane=issue&itemId=19926250)

/this test
 - [ ] `<Requests />` renders in the error state when applicable
 
 the only way this page renders in the error state is when there is an api related error, because of this line: https://github.com/scientist-softserv/webstore/blob/9d3b31cbc01224f57ccac8a685ad49831ae8977a/pages/requests/index.js#L23
 
 so i thought the best way to test it would be to mock an api error. but lmk if you think there is a better way

Reply:
ahh. true. can we make this mimic a 500 error instead? that's more likely than a 404 in this use case. there could be a 500 error fixture that's just an object that returns a 500 status and a message. then that can be imported here and used as the `error` value instead of "true". (again, more closely resembling what would actually happen in the code.)

Reply:
what do you mean by "that can be imported here"?

I don't think there is a way to use the statusCode value as the value for error in the test, because that intercept block will never run unless error is already true or truthy

Reply:
also it doesn't work to use a fixture in place of a status code and error message while intercepting- you need to pass the statusCode directly in order for the intercept to throw an error

Reply:
i changed it to a 500 error though which still worked